aiohttp_sse/helpers.py

```diff
@@ -1,43 +1,62 @@
-from collections.abc import Coroutine
+from types import TracebackType
+from typing import (
+    Any,
+    AsyncContextManager,
+    Coroutine,
+    Generator,
+    Optional,
+    Type,
+    TypeVar,
+)
 
+T = TypeVar("T", bound=AsyncContextManager["T"])
 
-class _ContextManager(Coroutine):
+
+class _ContextManager(Coroutine[T, None, T]):
     __slots__ = ("_coro", "_obj")
 
-    def __init__(self, coro):
+    def __init__(self, coro: Coroutine[T, None, T]) -> None:
         self._coro = coro
-        self._obj = None
+        self._obj: Optional[T] = None
 
-    def send(self, arg):
+    def send(self, arg: Any) -> T:
         return self._coro.send(arg)  # pragma: no cover
 
-    def throw(self, arg):
-        return self._coro.throw(arg)  # pragma: no cover
+    def throw(self, *args) -> T:  # type: ignore[no-untyped-def]
+        return self._coro.throw(*args)  # pragma: no cover
 
-    def close(self):
+    def close(self) -> None:
         return self._coro.close()  # pragma: no cover
 
-    @property
-    def gi_frame(self):
-        return self._coro.gi_frame  # pragma: no cover
-
-    @property
-    def gi_running(self):
-        return self._coro.gi_running  # pragma: no cover
-
-    @property
-    def gi_code(self):
-        return self._coro.gi_code  # pragma: no cover
-
-    def __next__(self):
+    # TODO: Do we really need it? Tests passes without it
+    # @property
+    # def gi_frame(self):
+    #     return self._coro.gi_frame  # pragma: no cover
+    #
+    # @property
+    # def gi_running(self):
+    #     return self._coro.gi_running  # pragma: no cover
+    #
+    # @property
+    # def gi_code(self):
+    #     return self._coro.gi_code  # pragma: no cover
```

Comment:
Mypy said that `Coroutine` has no such properties
Tests passes without it
Do we really need this overrides?

Reply:
I think, same as my other comment, ignore the errors and we should remove the class later. Likely this tries to provide some compatibility with internal coroutine objects.